pkg/oci/builder.go

```diff
@@ -269,3 +267,22 @@ func updateLastLink(cfg *buildConfig) error {
 	_ = os.RemoveAll(cfg.lastLink())
 	return os.Symlink(cfg.buildDir(), cfg.lastLink())
 }
+
+type testHelper struct {
+	emulateSlowBuild bool
+	continueCh       chan any
+
+	notifyDone bool
+	doneCh     chan any
+
+	notifyPaused bool
+	pausedCh     chan any
+}
+
+func newTestHelper() *testHelper {
+	return &testHelper{
+		continueCh: make(chan any),
+		doneCh:     make(chan any),
+		pausedCh:   make(chan any),
+	}
+}
```

Comment:
I think I saw a comment of yours yesterday about adding test structures in the main code. Can you help me understand why this is necessary?

Reply:
Yes, I did make a comment about this yesterday.   Testing concurrent build arbitration logic requires forcing one of two or more build tasks to go slow enough to cause a collision with the other.  To do this without creating a racing condition is nicely handled using channels.

To do this while keeping everthing test-related out of the core code would, in my opinion, be much more difficult to understand than the test helper with a few channels here.

So, in the end, I decided to try this, favoring simplicity